BootloaderCommonPkg/Library/TpmLib/TpmLib.c

```diff
@@ -17,11 +17,72 @@
 #include <Library/PcdLib.h>
 #include <Pi/PiBootMode.h>
 #include <IndustryStandard/Tpm2Acpi.h>
+#include <Library/SecureBootLib.h>
 #include "Tpm2CommandLib.h"
 #include "Tpm2DeviceLib.h"
 #include "TpmLibInternal.h"
 #include "TpmEventLog.h"
 
+
+
+/**
+  Get Tcg PCR Hash bank type required by bootloader.
+
+  @retval TpmHashAlg    TCG Algorithm Registry.
+**/
+UINT32
+GetTcgHashEnabled (
+  VOID
+  )
+{
+  UINT32 TpmHashAlg = 0;
+  UINT8  PcdHashType;
+
+  PcdHashType = PcdGet8(PcdMeasuredBootHashType);
+
+  if(PcdHashType == HASH_TYPE_SHA256){
+    TpmHashAlg = HASH_ALG_SHA256;
+  } else if (PcdHashType == HASH_TYPE_SHA384){
+    TpmHashAlg = HASH_ALG_SHA384;
+  } else if (PcdHashType == HASH_TYPE_SHA512){
+    TpmHashAlg = HASH_ALG_SHA512;
+  } else if (PcdHashType == HASH_TYPE_SM3){
+    TpmHashAlg = HASH_ALG_SM3_256;
+  }
+
+  return TpmHashAlg;
+}
+
+
+/**
+  Get Tpm Hash Algo type enabled.
+
+  @retval PcdHashType    TPM Algorithm Id.
+**/
+UINT32
+GetTpmHashAlgEnabled (
+  VOID

```

Comment:
Better let this function return TPMI_ALG_HASH or HASH_ALG_TYPE instead of UINT32?

Reply:
Although SBL does NOT support multiple active PCR banks, there may be use-case when customer wants to support that. To keep our code scalable, let's keep this as UINT32 itself to reflect a bit mask.

Reply:
This function would be updated to support MeasuredBoot Hashmask .